cmd/runtime_options_test.go

```diff
@@ -24,246 +24,255 @@ import (
-	for _, tc := range envVarTestCases {
-		t.Run(fmt.Sprintf("EnvVar test '%s'", tc.name), func(t *testing.T) {
-			os.Clearenv()
-			for key, val := range tc.systemEnv {
-				require.NoError(t, os.Setenv(key, val))
-			}
-			flags := runtimeOptionFlagSet(tc.useSysEnv)
-			require.NoError(t, flags.Parse(tc.cliOpts))
+func testRuntimeOptionsCase(t *testing.T, tc runtimeOptionsTestCase) {
+	flags := runtimeOptionFlagSet(tc.useSysEnv)
+	require.NoError(t, flags.Parse(tc.cliFlags))
+
+	rtOpts, err := getRuntimeOptions(flags, tc.systemEnv)
+	if tc.expErr {
+		require.Error(t, err)
+		return
+	}
+	require.NoError(t, err)
+	require.EqualValues(t, tc.expEnv, rtOpts.Env)
+	assert.Equal(t, tc.expCompatMode, rtOpts.CompatibilityMode)
 
-			rtOpts, err := getRuntimeOptions(flags)
-			if tc.expErr {
-				require.Error(t, err)
-				return
-			}
-			require.NoError(t, err)
-			require.EqualValues(t, tc.expEnv, rtOpts.Env)
+	compatMode, err := lib.ValidateCompatibilityMode(rtOpts.CompatibilityMode.String)
+	require.NoError(t, err)
 
-			// Clear the env again so real system values don't accidentally pollute the end-to-end test
-			os.Clearenv()
+	jsCodeLines := make([]string, 0, len(tc.expEnv)+2)
+	if compatMode == lib.CompatibilityModeExtended {
+		jsCodeLines = append(jsCodeLines, "export default function() {")
+	} else {
+		jsCodeLines = append(jsCodeLines, "module.exports.default = function() {")
+	}
 
-			jsCode := "export default function() {\n"
-			for key, val := range tc.expEnv {
-				jsCode += fmt.Sprintf(
-					"if (__ENV.%s !== `%s`) { throw new Error('Invalid %s: ' + __ENV.%s); }\n",
-					key, val, key, key,
-				)
-			}
-			jsCode += "}"
+	for key, val := range tc.expEnv {
+		jsCodeLines = append(jsCodeLines, fmt.Sprintf(
+			"if (__ENV.%s !== `%s`) { throw new Error('Invalid %s: ' + __ENV.%s); }",
+			key, val, key, key,
+		))
+	}
+	jsCodeLines = append(jsCodeLines, "}")
+	jsCode := []byte(strings.Join(jsCodeLines, "\n"))
 
-			// windows requires the environment variables to be loaded to gerenate the rand source
-			if runtime.GOOS == "windows" {
-				for _, e := range envVars {
-					parts := strings.Split(e, "=")
-					os.Setenv(parts[0], parts[1])
-				}
-			}
+	fs := afero.NewMemMapFs()
+	require.NoError(t, afero.WriteFile(fs, "/script.js", jsCode, 0644))
+	runner, err := newRunner(
+		&loader.SourceData{Data: jsCode, URL: &url.URL{Path: "/script.js", Scheme: "file"}},
+		typeJS,
+		map[string]afero.Fs{"file": fs},
+		rtOpts,
+	)
+	require.NoError(t, err)
 
-			fs := afero.NewMemMapFs()
-			require.NoError(t, afero.WriteFile(fs, "/script.js", []byte(jsCode), 0644))
-			runner, err := newRunner(
-				&loader.SourceData{
-					Data: []byte(jsCode),
-					URL:  &url.URL{Path: "/script.js", Scheme: "file"},
-				},
-				typeJS,
-				map[string]afero.Fs{"file": fs},
-				rtOpts,
-			)
-			require.NoError(t, err)
+	archive := runner.MakeArchive()
+	archiveBuf := &bytes.Buffer{}
+	assert.NoError(t, archive.Write(archiveBuf))
```

Comment:
I think this should be `require` as if this fails the below ones will definitely fail and we won't get any useful information from their failure

Reply:
Fixed by https://github.com/loadimpact/k6/pull/1325/commits/c9d75bd1cb74cafd11182125da5ffe2d8e61b37b